service/service.go

```diff
@@ -0,0 +1,25 @@
+package service
+
+import (
+	"fmt"
+)
+
+type Service struct {
+	ch chan string
+}
+
+func NewService(ch chan string) *Service {
+	return &Service{ch: ch}
+}
+
+func (s *Service) SetUpstream(host string) {
+	s.ch <- fmt.Sprintf(`kong.response.set_upstream:%s`, host)
```

Comment:
how should we handle functions like this one that can return `nil, err`? Go has multiple returns, so I guess `(bool, error)` would be the correct idiomatic pattern for these?